shop/src/main/resources/application.properties

```diff
@@ -1,8 +1,23 @@
 #spring.flyway.enabled=false
-spring.datasource.url=jdbc:h2:~/test;DB_CLOSE_ON_EXIT=FALSE
+spring.datasource.url=jdbc:h2:mem:test;DB_CLOSE_ON_EXIT=FALSE;
```

Comment:
So this is the application.properties file (should not be used for testing). So here it should be the real DB. And the in-memory DB should be used in application-integrationtest.properties. You just switched that. 
Then, on your test class, you specify the properties to use with @TestPropertySource annotation.

Reply:
Now I think I understood and fixed